gdx/test/com/badlogic/gdx/utils/QueueTest.java

```diff
@@ -374,6 +402,233 @@ public void hashEqualsTest () {
 		}
 	}
 
+	@Test
+	public void emptyQueueLast () {
+		Queue<Integer> queue = new Queue<>();
+
+		// Queue is empty and an error should be thrown.
+		exception.expect(NoSuchElementException.class);
+		queue.last();
+	}
+
+	@Test
+	public void emptyQueueRemoveFirst () {
+		Queue<Integer> queue = new Queue<>();
+
+		// Queue is empty and an error should be thrown.
+		exception.expect(NoSuchElementException.class);
+		queue.removeFirst();
+	}
+
+	@Test
+	public void emptyQueueRemoveLast () {
+		Queue<Integer> queue = new Queue<>();
+
+		// Queue is empty and an error should be thrown.
+		exception.expect(NoSuchElementException.class);
+		queue.removeLast();
+	}
+
+	@Test
+	public void emptyQueueFirst () {
+		Queue<Integer> queue = new Queue<>();
+
+		// Queue is empty and an error should be thrown.
+		exception.expect(NoSuchElementException.class);
+		queue.first();
+	}
+
+	@Test
+	public void indexOfNotFound () {
+		Queue<Integer> queue = new Queue<>();
+
+		// Check an existing element
+		queue.addFirst(1);
+		assertEquals(0, queue.indexOf(1, true));
```

Comment:
you shouldn't use identity comparison with Integer class. (same remark for some other lines below)